code/game/atom/atom_appearance.dm

```diff
@@ -56,6 +56,8 @@
 				continue
 			if(istext(maybe_not_an_atom) || isicon(maybe_not_an_atom))
 				continue
+			if (cached_color_filter && !(appearance_flags & KEEP_TOGETHER))
+				maybe_not_an_atom.add_filter(FILTER_ATOM_PRIORITY_COLOR, 1, cached_color_filter)
```

Comment:
this is done before maybe_not_an_atom is copied, which risks corrupting possible cached overlays